### DISCUSSION TITLE
custom script- checking if value in query_params is null

### DISCUSSION THREAD

**Discussion by el-magneto** [netbox-community/netbox#8677] [Q&A]
Hi,
I'm building custom script and would like to see  only circuits that are not connected.

```
    isp_primary  = ObjectVar(
            model = Circuit,
            required=True,
            query_params={
                "termination_a": null,
                 "termination_z": null,
                  }
            )
```
but i get:
```
<class 'NameError'>

name 'null' is not defined

Python version: 3.9.7
NetBox version: 3.1.6
```
Is there any way to filter this field on null parameters?

**Comment by hagbarddenstore** [comment]
Replace `null` with `None`.

Pythons `null` type is called `None`. It will be converted into the correct query parameter.

**Reply by el-magneto** [reply]
With "None" there is no failure, but it shows me all possible circuits - also those with "termination_a" defined. Maybe there is some trick nedded because value of "termination_a" is a dictionary?

**Reply by el-magneto** [reply]
Another question is how can I reach deeper values in query_params, like "termination_a->site->id"  here:
 ```
"termination_a": {
        "id": 15,
        "url": "http://ipam-test.gnb.adm/api/circuits/circuit-terminations/15/",
        "display": "Termination A: Kat Kolo 22",
        "site": {
            "id": 34,
```
Is it possible?

**Reply by hagbarddenstore** [reply]
Hm guess I'm wrong... Try `'null'`

**Reply by el-magneto** [reply]
'null' doesn't work

**Comment by el-magneto** [comment]
Things like: 'null', null, None, '',  doesn't work. I have no other ideas...

**Comment by jeremystretch** [comment]
Where did you see `termination_a` and `termination_z` listed as supported filters? AFAICT there's no logic in place for that, and they're not listed in the REST API documentation.

**Comment by el-magneto** [comment]
So correct me if I'm wrong. 
If "termination_a" value was an empty string I could use ' "termination_a__empty": True' statement, but because its value is a dictionary or null there is no way to query that?
